File: Jardin De infantes/EE/Servicios/Encriptado/EncriptarSHA256.vb

Imports System.Security.Cryptography
Imports System.Text
Public Class EncriptarSHA256
    Inherits eeEncriptado
    Sub New(_valorInicial As String)
        Me.valorInicial = _valorInicial


    End Sub
    Public Overrides Function Hashear() As String
        Dim SHA256 As SHA256 = SHA256Managed.Create()
        Dim bytes As Byte() = Encoding.UTF8.GetBytes(Me.valorInicial)
        Dim hash As Byte() = SHA256.ComputeHash(bytes)
        Dim stringBuilder As New StringBuilder()

        For i As Integer = 0 To hash.Length - 1
            stringBuilder.Append(hash(i).ToString("X2"))
        Next

        Return stringBuilder.ToString
    End Function
End Class

File: Jardin De infantes/Seguridad/EEseguridad/Encriptado/EncriptarSHA256.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Security.Cryptography;


namespace EEseguridad.Encriptado
{
   public class EncriptarSHA256 : eeEnciptador
        
    {
       public EncriptarSHA256(string _valorInicial)
        {
            valorInicial = _valorInicial;
        }

        public override string Hashear()
        {
            SHA256 SHA256 = SHA256Managed.Create();
            byte[] bytes = Encoding.UTF8.GetBytes(this.valorInicial);
            byte[] hash = SHA256.ComputeHash(bytes);
            StringBuilder stringBuilder = new StringBuilder();
            for (int i = 0; i <= hash.Length - 1; i++)
            {
                stringBuilder.Append(hash[i].ToString("X2"));
            }

            return stringBuilder.ToString();
        }
    }
}
